Fix "precedes" listchar behavior in wrap mode

zwegner opened the pull request:
Previously, the "precedes" character would be rendered on every row
when w_skipcol > 0 (i.e., when viewing a single line longer than the
entire screen), instead of just on the first row. Make sure to only
render it on the first row in this case.

Add a test for this behavior.

Fix documentation for the "precedes" character, which erroneously
stated that it was only active when wrap mode was off.

zwegner committed:
Fix "precedes" listchar behavior in wrap mode

Previously, the "precedes" character would be rendered on every row
when w_skipcol > 0 (i.e., when viewing a single line longer than the
entire screen), instead of just on the first row. Make sure to only
render it on the first row in this case.

Add a test for this behavior.

Fix documentation for the "precedes" character, which erroneously
stated that it was only active when wrap mode was off.

chrisbra commented:
that applies upstream as well. cc @brammool

brammool commented:
> that applies upstream as well. cc @brammool

Well, can someone make a pull request for Vim?

Not sure how to make the "precedes" character show up when 'wrap' is
set.

-- 
   They now pass three KNIGHTS impaled to a tree.  With their feet off the
   ground,  with one lance through the lot of them, they are skewered up
   like a barbecue.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

zwegner commented:
> Not sure how to make the "precedes" character show up when 'wrap' is set.

Try `:set list listchars=precedes:< wrap | normal 10000ia`

chrisbra commented:
> Well, can someone make a pull request for Vim?

I'll go ahead and do it and create a test.

chrisbra commented:
see vim/vim#4953

chrisbra commented:
@zwegner thanks for tracking this. I have actually noted this behaviour several times for overlong lines in a window, but never time to fix this.

blueyed commented:
@zwegner 
Thanks indeed!

Let's wait for the Vim patch here then (via https://github.com/vim/vim/pull/4953), which could be force-pushed it here then.

justinmk committed:
vim-patch:8.1.2060: "precedes" in 'listchars' not used properly

(Credit: Zach Wegner, https://github.com/neovim/neovim/pull/11034)

Problem:    "precedes" in 'listchars' not used properly.
Solution:   Correctly handle the "precedes" char in list mode for long lines.
https://github.com/vim/vim/commit/bffba7f7042f6082e75b42484b15f66087b01941

wsdjeg commented:
> Let's wait for the Vim patch here then (via vim/vim#4953), which could be force-pushed it here then.

@blueyed @zwegner vim patch 8.1.2060 do not include the origin patch author Zach Wegner, and should not be force-pushed. why we need to wait for vim patch if it does not include the author info of this PR?

bfredl commented:
@wsdjeg if you look at the actual merged commits this was already handled as good it possibly could be..
